Game/Scripts/LightFinisherAttackScript.cpp

```diff
@@ -0,0 +1,89 @@
+#include "StdAfx.h"
+#include "LightFinisherAttackScript.h"
+
+#include "Application.h"
+
+#include "Scripting/ScriptFactory.h"
+#include "ModuleInput.h"
+#include "ModuleScene.h"
+#include "Scene/Scene.h"
+
+#include "Components/ComponentScript.h"
+
+#include "../Scripts/LightAttackBullet.h"
+#include "../Scripts/EntityDetection.h"
+
+REGISTERCLASS(LightFinisherAttackScript);
+
+LightFinisherAttackScript::LightFinisherAttackScript() 
+	: Script(), 
+	loadedScene(nullptr), 
+	bulletPrefab(nullptr),
+	bulletVelocity(15.0f),
+	stunTime(2.0f),
+	enemyDetection(nullptr),
+	enemyDetectionObject(nullptr),
+	currentCooldown(0.0f),
+	cooldown(2.0f)
+{
+	REGISTER_FIELD(bulletPrefab, GameObject*);
+	REGISTER_FIELD(bulletVelocity, float);
+	REGISTER_FIELD(stunTime, float);
+	REGISTER_FIELD(enemyDetectionObject, GameObject*);
```

Comment:
I'm pretty sure you could set the type of the field to the type of the script you want. You don't seem to be using this GameObject for anything other than extracting the script.
```suggestion
	REGISTER_FIELD(enemyDetection, EntityDetection*);
```

Reply:
Done thanx